envoy/config/filter/accesslog/v2/accesslog.proto

```diff
@@ -358,16 +361,32 @@ message NotHealthCheckFilter {
 message TraceableFilter {
 }
 
-// Filters for random sampling of requests. Sampling pivots on the header
-// :ref:`x-request-id<config_http_conn_man_headers_x-request-id>` being present. If
-// :ref:`x-request-id<config_http_conn_man_headers_x-request-id>` is present, the filter will
-// consistently sample across multiple hosts based on the runtime key value and the value extracted
-// from :ref:`x-request-id<config_http_conn_man_headers_x-request-id>`. If it is missing, the
-// filter will randomly sample based on the runtime key value.
+// Filters for random sampling of requests.
 message RuntimeFilter {
-  // Runtime key to get the percentage of requests to be sampled. This runtime
-  // control is specified in the range 0-100 and defaults to 0.
+  // Runtime key to get the percentage of requests to be sampled.
   string runtime_key = 1 [(validate.rules).string.min_bytes = 1];
+
+  // The default sampling rate if a value is not found in runtime. Defaults to 0.
+  uint32 default = 2;
+
+  // The sampling rate divisor. This field is used in conjunction with *default* to determine the
+  // final sampled percentage. For example, setting this field to 100 allows sampling in steps of
+  // 1%. Setting this field to 10000 allows sampling in steps of 0.01%. The default if not
+  // specified is 100.
+  google.protobuf.UInt32Value divisor = 3 [(validate.rules).uint32.gt = 0];
+
+  // By default, sampling pivots on the header
+  // :ref:`x-request-id<config_http_conn_man_headers_x-request-id>` being present. If
+  // :ref:`x-request-id<config_http_conn_man_headers_x-request-id>` is present, the filter will
+  // consistently sample across multiple hosts based on the runtime key value and the value
+  // extracted from :ref:`x-request-id<config_http_conn_man_headers_x-request-id>`. If it is
+  // missing, or *use_independent_randomness* is set to true, the filter will randomly sample based
+  // on the runtime key value alone. *use_independent_randomness* can be used for logging kill
+  // switches within complex nested :ref:`AndFilter
```

Comment:
Might want to say something along the lines of:

> Set this to true if you need this filter to behave like an independent random variable when composed with logical operator filters.

Reply:
added more text.